packages/go_router/test/inherited_go_router_test.dart

```diff
@@ -0,0 +1,77 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:go_router/go_router.dart';
+import 'package:go_router/src/inherited_go_router.dart';
+
+void main() {
+  test('does not update on changes', () {
```

Comment:
We should also add a test for the opposite case. InheritedGoRouter's updateShouldNotify always returns false, but it should probably be checking that it's `goRouter` field is the same as `oldWidget`'s `goRouter`.